src/Adapter/Customer/CommandHandler/AddCustomerHandler.php

```diff
@@ -83,7 +84,13 @@ public function handle(AddCustomerCommand $command)
             throw new CustomerException('Customer contains invalid field values');
         }
 
-        $this->assertCustomerWithGivenEmailDoesNotExist($command->getEmail());
+        // If dealing with a registered customer, we need to check if the email does not exist.
+        // Two guests with the same email can co-exist, two registered customers can not.
+        if (!$command->isGuest()) {
```

Comment:
What about a guest that uses the same email that a registered user ?

Reply:
That is allowed, thats why I put it inside the condition.